### DISCUSSION TITLE
Getting the SUMO Vehicle Type in the Artery Simulation

### DISCUSSION THREAD

**Discussion by Shaaban52** [riebl/artery#292] [Q&A]
Hello, 

Thank you in advance for your support.

I have described a traffic scenario in SUMO and defined some related information for vehicles, such as the Vehicle type. Then I used Artery for the V2X simulation. I successfully obtained the information from the CAM messages (e.g., message ID, Station Id, Station Type, etc.). 

However, I noticed that the StationID (from CAM data) does not match the vehicle type defined in SUMO. Now, I am trying to extract the vehicle type defined in SUMO from the Artery simulation, but it seems to be a challenge for me!!!

Are there any ways in Artery that I can get vehicle infromation (defined in SUMO), specifically the vehicle type?

FYI, I am using the latest Artery version.

Many thanks and regards

**Comment by riebl** [comment]
Hi @Shaaban52,

the *StationID* field in CAM headers has no link to the vehicle type, but the *StationType* in the *basic* CAM container has. At the moment, it is always set to "passenger car" here: https://github.com/riebl/artery/blob/4bc468e372091fa392d43c5fc3358821a21534c9/src/artery/application/CaService.cc#L222
It should be not too difficult to change this behaviour because the `VehicleDataProvider` has already a `getStationType` getter. You just need to set the CAM field depending on this function's return value.

**Reply by Shaaban52** [reply]
Hi Riebl, 

Thanks a lot for your replay. 

It is clear to me now, but what is currently challenging me is how to obtain the vehicle type from SUMO. Do you have any documentation or suggestions that I can follow to retrieve the vehicle type from SUMO?

Regards

**Reply by riebl** [reply]
Hi @Shaaban52,

the station type used for initialization of `VehicleDataProvider` comes ultimately from SUMO's vehicle class: https://github.com/riebl/artery/blob/4bc468e372091fa392d43c5fc3358821a21534c9/src/traci/sumo/utils/traci/TraCIAPI.h#L638C21
We have a dedicated function in Artery which maps SUMO vehicle classes to ITS station types: https://github.com/riebl/artery/blob/4bc468e372091fa392d43c5fc3358821a21534c9/src/artery/application/StationType.cc#L17
